core/src/main/java/org/elasticsearch/index/analysis/CustomAnalyzerProvider.java

```diff
@@ -88,6 +94,34 @@ public void build(final Map<String, TokenizerFactory> tokenizers, final Map<Stri
         );
     }
 
+    public static TokenFilterFactory checkAndApplySynonymFilter(TokenFilterFactory tokenFilter, String tokenizerName, TokenizerFactory tokenizer,
+                                                                List<TokenFilterFactory> tokenFilterList,
+                                                                List<CharFilterFactory> charFiltersList,
+                                                                int positionIncrementGap, int offsetGap, Environment env) {
```

Comment:
What I mean is that gaps only matter when there are multiple field values. In that case, the analyzer will always analyze a single value at a time, so the value of the gaps do not matter. We can pass whatever value we want and clean up the signature of this method.